payment/src/main/java/org/killbill/billing/payment/core/PaymentProcessor.java

```diff
@@ -410,11 +410,6 @@ public boolean apply(final PaymentTransactionModelDao input) {
             currentStateName = paymentModelDao.getLastSuccessStateName();
         }
 
-        // Sanity: no paymentMethodId was passed through API and account does not have a default paymentMethodId
-        if (paymentStateContext.getPaymentMethodId() == null) {
-            throw new PaymentApiException(ErrorCode.PAYMENT_NO_DEFAULT_PAYMENT_METHOD, paymentStateContext.getAccount().getId());
```

Comment:
@pierre @sbrossie 

> In JaxRsResourceBase#validatePaymentMethodForAccount, instead of verifyNonNull, can you throw this ErrorCode?

I'll replace the **verifyNonNull** method throwing the **PaymentApiException** using the **PAYMENT_NO_DEFAULT_PAYMENT_METHOD** Error Code. Is that what we are expecting?